aten/src/ATen/native/sparse/Hardshrink.cpp

```diff
@@ -0,0 +1,112 @@
+#include <ATen/ATen.h>
+#include <ATen/Config.h>
+#include <ATen/NamedTensorUtils.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/Parallel.h>
+#include <ATen/SparseTensorUtils.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+#include <map>
+
+namespace at {
+namespace native {
+namespace {
+
+void sparse_hardshrink_kernel(TensorIterator& iter, Scalar lambd) {
+  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "sparse_hardshrink_cpu", [&] {
+    auto lambd_val = lambd.to<scalar_t>();
+    cpu_kernel_vec(
+        iter,
+        [=](scalar_t self_val) {
+          return (self_val >= -lambd_val && self_val <= lambd_val) ? scalar_t(0)
+                                                                   : self_val;
+        },
+        [=](Vec256<scalar_t> self_val) {
+          return ((self_val < -lambd_val) | (self_val > lambd_val)) & self_val;
```

Comment:
If I understand sparse iterators correctly, the second condition shrinks the indices, correct?

Reply:
Yes, the pytorch tensor iterators for CPU has two lambdas. Once for simple scalar processing and the other for vector processing. Both shrinks the values depending on the `lambd` input.